pkg/acs/calacs/acsccd/blev_funcs_postsm4.c

```diff
@@ -36,152 +39,218 @@ static int make_amp_array(const int arr_rows, const int arr_cols, SingleGroup *
 static int unmake_amp_array(const int arr_rows, const int arr_cols, SingleGroup * im,
                             int amp, double * array);
 
+/* Remove the signal dependent bias shift from post-SM4 full frame WFC data
+ * (based on ISR http://www.stsci.edu/hst/acs/documents/isrs/isr1202.pdf),
+ * as well as from supported subarray data (~post-mid-2016) as identified by
+ * specific aperture names 
+ * (based on ISR http://www.stsci.edu/hst/acs/documents/isrs/isr1703.pdf).
+ *
+ * chip2 consists of amps C & D, chip1 consists of amps A & B. 
+ *
+ * Because full frame data uses both chips, but supported subarray data
+ * is only from a single amp (therefore, a portion of a single chip), this
+ * routine has a variable number of input SingleGroups.
+ *
+ * 16-May-2018 M.D. De La Pena: Generalized bias_shift_corr() to handle subarray data.
+ */
+int bias_shift_corr(ACSInfo *acs, int nGroups, ...) {
```

Comment:
Add ``assert(acs);``.

Reply:
Also:
```c
if (nGroups < 1)
    return HSTCAL_OK; // or NOTHING_TO_DO, NO_GOOD_DATA, INTERNAL_ERROR
```

Reply:
Actually, if going for ``INTERNAL_ERROR`` I would instead ``assert``. This is a much better way to handle internal errors as it gives the src line in question.

Reply:
I tend to treat this kind of thing as a NOP and would return ``HSTCAL_OK``,  upto you to think about the function's semantics.

Reply:
Yes.